clients/cobol-lsp-vscode-extension/src/services/LanguageClientService.ts

```diff
@@ -13,30 +13,40 @@
  */
 
 import * as fs from "fs";
+import * as os from "os";
 import * as net from "net";
 import * as vscode from "vscode";
+import * as cp from "child_process";
 import {
     ConfigurationParams,
     ConfigurationRequest,
+    Executable,
     LanguageClient,
     LanguageClientOptions,
     StreamInfo,
 } from "vscode-languageclient";
 import {ConfigurationWorkspaceMiddleware} from "vscode-languageclient/lib/configuration";
+import {GenericNotificationHandler, GenericRequestHandler} from "vscode-languageserver-protocol";
 import {LANGUAGE_ID} from "../constants";
 import {JavaCheck} from "./JavaCheck";
 import {Middleware} from "./Middleware";
-import {GenericNotificationHandler, GenericRequestHandler} from "vscode-languageserver-protocol";
 import { SettingsService } from "./Settings";
 
 export class LanguageClientService {
-    private readonly jarPath: string;
+    private jarPath: string;
     private languageClient: LanguageClient;
     private handlers: {(languageClient: LanguageClient): void}[] = [];
+    private isNativeBuildEnabled: boolean = false;
 
     constructor(private middleware: Middleware) {
         const ext = vscode.extensions.getExtension("BroadcomMFD.cobol-language-support");
-        this.jarPath = `${ext.extensionPath}/server/server.jar`;
+        this.jarPath = `${ext.extensionPath}/server/jar/server.jar`;
+    }
+
+    public enableNativeBuild() {
+        const ext = vscode.extensions.getExtension("BroadcomMFD.cobol-language-support");
+        this.isNativeBuildEnabled = true;
+        this.jarPath = this.initializeExecutables(`${ext.extensionPath}/server`);
```

Comment:
I think that jarPath is not a good name anymore

Reply:
I agree, will update it.

Reply:
updated